pkg/storegateway/bucket_stores.go

```diff
@@ -551,17 +559,31 @@ func (u *BucketStores) closeBucketStoreAndDeleteLocalFilesForExcludedTenants(inc
 	}
 }
 
-// getBlocksLoadedMetric returns the number of blocks currently loaded across all bucket stores.
-func (u *BucketStores) getBlocksLoadedMetric() float64 {
-	count := 0
+func (u *BucketStores) Describe(descs chan<- *prometheus.Desc) {
+	descs <- u.blocksLoaded
+	descs <- u.blocksLoadedByDuration
+}
+
+func (u *BucketStores) Collect(metrics chan<- prometheus.Metric) {
+	loaded := make(map[time.Duration]int)
+	total := 0
 
 	u.storesMu.RLock()
+
 	for _, store := range u.stores {
-		count += store.Stats().BlocksLoaded
+		stats := store.Stats(u.cfg.TSDB.BlockRanges)
+		for d, n := range stats.BlocksLoaded {
+			loaded[d] += n
+			total += n
+		}
 	}
+
 	u.storesMu.RUnlock()
 
-	return float64(count)
+	metrics <- prometheus.MustNewConstMetric(u.blocksLoaded, prometheus.GaugeValue, float64(total))
+	for d, n := range loaded {
+		metrics <- prometheus.MustNewConstMetric(u.blocksLoadedByDuration, prometheus.GaugeValue, float64(n), d.String())
```

Comment:
Can we convert `time.Duration` to `model.Duration` and use its `String` method? It will remove unnecessary `0m0s` suffixes.

Reply:
Alternatively: we can change `duration` to `hours`, to avoid formatting `24h` as `1d` when using `model.Duration`. (Although I don't think that's a big deal)

Reply:
I'll convert to `model.Duration`. I was trying to think of a good way to remove the trailing `0s` without assuming that block durations were always multiples of hours.

Reply:
Switched to `model.Duration` for formatting of the label value in 8fa952c172f6028d6be259f3c5aeb7da7f4bc795